diff --git a/audio/out/ao_audiotrack.c b/audio/out/ao_audiotrack.c
--- a/audio/out/ao_audiotrack.c
+++ b/audio/out/ao_audiotrack.c
@@ -676,4 +676,7 @@ static int init(struct ao *ao)
-    if (!af_fmt_is_spdif(ao->format) && AudioTrack.getNativeOutputSampleRate) {
+    if (af_fmt_is_spdif(ao->format))
+        MP_VERBOSE(ao, "Using device native output sample rate for passthrough compatibility\n");
+
+    if (AudioTrack.getNativeOutputSampleRate) {
         jint samplerate = MP_JNI_CALL_STATIC_INT(
             AudioTrack.clazz,
             AudioTrack.getNativeOutputSampleRate,
